vm/vm.go

```diff
@@ -145,6 +145,19 @@ func (vm *Engine) validateSplit() error {
 		return err
 	}
 
+	// If the split requires a zero-value root asset, the root asset must
+	// be unspendable. Non-inflation of the split is enforced elsewhere, at
+	// the end of vm.Execute().
+	if vm.newAsset.Amount == 0 && !vm.newAsset.IsUnspendable() {
+		return newErrKind(ErrInvalidRootAsset)
+	}
+
+	// If we are validating the root asset of the split, the root split must
+	// also be unspendable.
+	if vm.splitAsset.Amount == 0 && !vm.splitAsset.IsUnspendable() {
```

Comment:
Not sure I understand this check, when would the _split_ itself have a zero value?

Reply:
Well we validate each split asset / leaf in the split tree, and one leaf represents the root asset. So on one of these `validateSplit()` calls, both `vm.NewAsset` and `vm.SplitAsset` will be representing the root asset. For that call, the split asset wouild also have zero value.

In the unspendable-root case, we should enforce the same conditions for both the root asset and its matching leaf in the split tree.

Reply:
Ah ok, I think this is the call site you're referring to: https://github.com/lightninglabs/taro/blob/main/taroscript/send.go#L447-L454

I think this is sort of a quirk of the way this particular validation is set up. In other words, this doesn't necessarily need to live here (thereby impacting the spec) if we either modify that loop to verify the root independently, or it omits passing the split when it's just the root. 

Otherwise, we'd be allowing a _zero value split_ in general scenarios outside of our usage in the `taroscript` package. I can't think of a good reason to allow that, other than that explicit vs implicit burn stuff we discussed in te past?